### ISSUE TITLE
Explore: LogQL parser misses labels in range aggregations

### ISSUE THREAD

**Issue by jtackaberry** [grafana/grafana#79194]
### What happened?

Suppose we have the given LogQL query:

```
quantile_over_time(0.95, {component="foobar"} |= "elapsed_time" | json | unwrap elapsed_time | __error__=""[1m]) by (host)
```

If we enter that query in the code view, it runs fine.

Now click the Builder button to visualize this LogQL in the query builder.   It's generally fine, _except_ that it misses the aggregation by `host` label:

![image](https://github.com/grafana/grafana/assets/1280551/323bb0b5-2eef-448a-beb8-3ca31fd3e26e)

### What did you expect to happen?

I expect it to parse out the aggregation by label and represent it in the query builder.  It should look like this:

![image](https://github.com/grafana/grafana/assets/1280551/e19fc3dd-3b0f-43a8-aafc-e866605cad00)

### Did this work before?

Don't recall.

### How do we reproduce it?

1. Open Explore view with a Loki data source
2. Enable code view, paste the LogQL example from above.
3. Click Builder

### Is the bug inside a dashboard panel?

No.

### Environment (with versions)?

Grafana: 10.2.2
OS: Linux (Grafana) and Windows (Firefox) 
Browser: Firefox 119

### Grafana platform?

Kubernetes

### Datasource(s)?

Loki v2.9.2

**Comment by jtackaberry** [comment]
> is this issue also reproducible in the query editor within a panel or is it specific to the explore query editor?

Just checked it out from the panel editor and yeah, same behavior there as well.

**Comment by matyax** [comment]
Thank you for reporting this @jtackaberry ! It will be fixed by https://github.com/grafana/grafana/pull/80145 and available in Grafana 10.3
